Assets/HoloToolkit/Common/Scripts/Singleton.cs

```diff
@@ -62,11 +72,48 @@ public static bool IsInitialized
             }
         }
 
+        public static bool ConfirmInitialized()
+        {
+            T access = Instance;
+            return IsInitialized;
+        }
+
+        [SerializeField]
+        private bool DontDestroyParentRootOnLoad = true;
+
+        private object initializedLock = new object();
+
+        protected void Initialize()
+        {
+            lock (initializedLock)
```

Comment:
What's the purpose of the lock here?

Reply:
In case an object is accessed on the main thread and another thread at the same time.  Definitely not common but since we have it might as well keep the protection.